### DISCUSSION TITLE
A new LGT8XM core RISC microcontroller with USB interface

### DISCUSSION THREAD

**Discussion by LaZsolt** [dbuezas/lgt8fx#285] [Show and tell]
Recently [Prodesign Semiconductor](http://www.prodesign.com.cn)'s made the PDS8F208A microcontroller with USB.
PDS8F208A has a built-in dual USB type-C port (can be used as PD fast charging protocol)

Some new, modified features compared to LGT8F328D:
- Built-in calibratable 24MHz RC oscillator, supports multiplier output for 48 MHz counting frequency
- 20 KB FLASH program memory
- 1.28V/2.56V high precision internal reference voltage
- 2 x UART interface
- 2 x USB A port
- 2 x USB C port PD controller
- 2 x 10-bit programmable DAC with 3mA output buffer
- 4 x 80mA high current push-pull drive IO
- DC/DC loop control

More detailed informations: [Click here](https://www-prodesign-com-cn.translate.goog/pds8f208a/?_x_tr_sch=http&_x_tr_sl=zh-CN&_x_tr_tl=en&_x_tr_hl=en&_x_tr_pto=sc)
Databook: http://www.prodesign.com.cn/wp-content/uploads/2023/06/PDS8F208A_dbook_pub_V1.1.5.pdf

It's looks like an upgraded LGT8F328D core. 
Block diagram:
![kép](https://github.com/dbuezas/lgt8fx/assets/60486174/896464a0-be68-41a3-9dd6-e04214f60c15)

**Comment by dbuezas** [comment]
That looks really interesting. The buffered dac and the two usb c pd controllers. I wonder why 2,

**Reply by LaZsolt** [reply]
I think this MCU is primarily designed to be used as USB charger or power bank controller.
With 2 DAC and  2 USB C conrollers can control two different charging voltage.

**Reply by dbuezas** [reply]
Oh, so one pd for the input and one for the output?
That's really neat, can these already be bought? It would be a really fun project to add support for it and make a home powerbank as example. With the timer clock doubler and two 80mAh outputs driving mosfets with dead time control driving a boost converter. And an ADC to close the loop. It's perfect!

**Comment by dwillmore** [comment]
Looks like the new WCH chip the CH32X035:
https://github.com/openwch/ch32x035

**Reply by trimarco232** [reply]
CH32X035 looks more like a STM32 , but with RISV-V core and 5 volts capable
it might be tricky to program it in arduino IDE , so wait and see

**Comment by dwillmore** [comment]
The ch32v003fun project might add support for it at some point.  Right now
it's a single header include library for bare metal programming of the
ch32V003, but support is being added for other members of the CH32 family.
Who knows where it will end up.  There's also a sort of arduino library
support for it as well.  There's more info on the ch32v003fun github of
chlor.  Come on over if you're interested.
